lib/semantics/mod-file.cc

```diff
@@ -192,19 +192,19 @@ void ModFileWriter::PutDerivedType(const Symbol &typeSymbol) {
 
 void ModFileWriter::PutSubprogram(const Symbol &symbol) {
   auto attrs{symbol.attrs()};
+  auto &details{symbol.get<SubprogramDetails>()};
   Attrs bindAttrs{};
   if (attrs.test(Attr::BIND_C)) {
     // bind(c) is a suffix, not prefix
     bindAttrs.set(Attr::BIND_C, true);
     attrs.set(Attr::BIND_C, false);
   }
-  bool isExternal{attrs.test(Attr::EXTERNAL)};
-  std::ostream &os{isExternal ? decls_ : contains_};
-  if (isExternal) {
+  bool isInterface{details.isInterface()};
+  std::ostream &os{isInterface ? decls_ : contains_};
+  if (isInterface) {
     os << "interface\n";
   }
   PutAttrs(os, attrs, ""s, " "s);
-  auto &details{symbol.get<SubprogramDetails>()};
```

Comment:
Is details used somewhere earlier? I don't see why it was moved up to the top of this member function.

Reply:
Yes, I added a call to `details.isInterface()` up a few lines.